### PR TITLE
decompose disc-compiler with org tensorflow

### PR DESCRIPTION
**Pull request by Yancey1989**
related issued #973 , please review https://github.com/pai-disc/tensorflow/pull/27 first.

PR summary:
- change visibility Bazel configuration, https://github.com/pai-disc/tensorflow/pull/27
- fix header file path from `tensorflow/compiler/mlir/disc/...` to `mlir/disc`
- make `tao_compiler` as a Bazel workspace
- make `tf_community` as a `local_repo` and update Bazel rule from `//tensorflow/...` to `@org_tensorflow//tensorflow/...` 
- remove sym link `tao_compiler` to `tf_community` in `tao_build.py` build script.

This PR also temporary disable `disc-replay` because of some linking error, and will re-open it ASAP.

### CODE REVIEW

File: tao_compiler/.bazelrc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+startup --host_jvm_args=-Djdk.http.auth.tunneling.disabledSchemes=
+
+try-import ../tf_community/.bazelrc
+try-import ../tf_community/.tf_configure.bazelrc
+try-import %workspace%/.bazelrc.user
+
+build --disk_cache=~/.cache
```
Comment (wyzero) [inline tao_compiler/.bazelrc:7]: Do we need disk cache or remote cache？

File: tao_compiler/mlir/disc/BUILD [hunk 2]
```diff
@@ -38,10 +39,10 @@ package_group(
         "//learning/brain/experimental/swift_mlir/...",
         "//learning/brain/google/xla/kernels/...",
         "//learning/brain/swift/swift_mlir/...",
-        "//tensorflow/compiler/mlir/...",
-        "//tensorflow/compiler/tf2xla/...",
-        "//tensorflow/compiler/xla/...",
-        "//tensorflow/compiler/...",
+        #"@org_tensorflow//tensorflow/compiler/mlir/...",
```
Comment (wyzero) [inline tao_compiler/mlir/disc/BUILD:42]: unused code？

File: tao_compiler/mlir/disc/BUILD [hunk 3]
```diff
@@ -2697,12 +2714,12 @@ gentbl_cc_library(
     ],
     tblgen = "@llvm-project//mlir:mlir-tblgen",
     td_file = "IR/disc_tf_additional_ops.td",
-    td_srcs = [
-        "//tensorflow/compiler/mlir/tensorflow:ir/tf_op_base.td",
-        "//tensorflow/compiler/mlir/tensorflow:ir/tf_op_interfaces.td",
-    ],
+    #td_srcs = [
```
Comment (wyzero) [inline tao_compiler/mlir/disc/BUILD:2717]: ditto

File: tao_compiler/mlir/disc/tests/BUILD [hunk 4]
```diff
@@ -27,10 +28,10 @@ package_group(
         "//learning/brain/experimental/swift_mlir/...",
         "//learning/brain/google/xla/kernels/...",
         "//learning/brain/swift/swift_mlir/...",
-        "//tensorflow/compiler/mlir/...",
-        "//tensorflow/compiler/tf2xla/...",
-        "//tensorflow/compiler/xla/...",
-        "//tensorflow/compiler/...",
+        "//mlir/...",
+        #"@org_tensorflow//tensorflow/compiler/tf2xla/...",
```
Comment (wyzero) [inline tao_compiler/mlir/disc/tests/BUILD:32]: ditto

File: tao_compiler/.bazelrc (hunk 1 above)
Comment (Yancey1989) [reply tao_compiler/.bazelrc:7]: according to the experience of torch-blade, disc-cache is more stable.

Comment (wyzero) [reply tao_compiler/.bazelrc:7]: I think remote-cache is stable enough (tao compiler uses remote cache without this PR) and can be GC automatically while disc cache can not be GC. also CC @qiuxiafei @tanyokwok

Comment (tanyokwok) [reply tao_compiler/.bazelrc:7]: The TorchBlade's ci script already deleted the legacy disk cache(older than 10 days each time) for a while. I must be stable enough for TorchBlade. FYI https://github.com/alibaba/BladeDISC/pull/662

Comment (Yancey1989) [reply tao_compiler/.bazelrc:7]: We can GC the disc-cache data using this code snippet: https://github.com/alibaba/BladeDISC/blob/main/scripts/ci/test_pytorch_blade.sh#L28-L38

But I agree that does not change the Bazel cache behavior in this PR.

File: pytorch_blade/bazel/torch/torch.BUILD.tpl [hunk 5]
```diff
@@ -15,7 +15,7 @@ cc_library(
             "lib/libtorch_cpu.so",
             "lib/libtorch_global_deps.so",
         ] + glob(
-            ["lib/libtorch_cuda.so","lib/libtensorpipe.so",]),
+            ["lib/libtorch_cuda_cu.so", "lib/libtorch_cuda_cpp.so", "lib/libgomp-a34b3233.so.1"]),
```
Comment (tanyokwok) [inline pytorch_blade/bazel/torch/torch.BUILD.tpl:18]: Nit: use wildcards for the suffix `lib/libgomp-xxxxxx.so.1`

File: pytorch_blade/pytorch_blade/compiler/mlir/runtime/ral_context.h [hunk 6]
```diff
@@ -30,11 +30,13 @@ using CUDAStream = ::c10::hip::HIPStream;
 #else // TORCH_BLADE_USE_ROCM
 #include <c10/cuda/CUDAStream.h>
 #endif // TORCH_BLADE_USE_ROCM
-#include "tensorflow/compiler/mlir/xla/ral/context/base/cuda/cuda_context_impl.h"
+//#include
+//"tensorflow/compiler/mlir/xla/ral/context/base/cuda/cuda_context_impl.h"
```
Comment (tanyokwok) [inline pytorch_blade/pytorch_blade/compiler/mlir/runtime/ral_context.h:34]: Nit: remove unused

File: pytorch_blade/scripts/build_pytorch_blade.sh [hunk 7]
```diff
@@ -53,7 +53,8 @@ function ci_build() {
       python3 ../scripts/python/common_setup.py --cpu_only $COMMON_SETUP_ARGS
     fi
     TORCH_LIB=$(python -c 'import torch; import os; print(os.path.dirname(os.path.abspath(torch.__file__)) + "/lib/")') \
-    export LD_LIBRARY_PATH=$TORCH_LIB:$LD_LIBRARY_PATH \
+    export LD_LIBRARY_PATH=$TORCH_LIB:$LD_LIBRARY_PATH
+    rm -rf ../tf_community/.tf_configure.bazelrc
```
Comment (tanyokwok) [inline pytorch_blade/scripts/build_pytorch_blade.sh:57]: Why is the line required?

File: scripts/ci/build_and_test.sh [hunk 8]
```diff
@@ -56,7 +56,7 @@ fi
 # copy Python wheel package to build folder
 mkdir -p build && \
 cp tao/dist/blade_disc*.whl ./build
-cp tf_community/bazel-bin/tensorflow/compiler/mlir/disc/tools/disc-replay/disc-replay-main ./build/
+#cp tao_compiler/bazel-bin/mlir/disc/tools/disc-replay/disc-replay-main ./build/
```
Comment (tanyokwok) [inline scripts/ci/build_and_test.sh:59]: remove unused

File: pytorch_blade/scripts/build_pytorch_blade.sh (hunk 7 above)
Comment (Yancey1989) [reply pytorch_blade/scripts/build_pytorch_blade.sh:57]: Sometimes the CI task can not GC the `.tf_configure.bazelrc`, the previous file will cause some unexpected errors.

Comment (Yancey1989) [reply pytorch_blade/scripts/build_pytorch_blade.sh]: I updated this PR and using `tao_build.py` to setup building env, so I deleted `rm -rf .tf_configure.bazelrc `.

**Review by tanyokwok — DISMISSED** [review]
LGTM

**Review by tanyokwok — APPROVED** [review]

**Comment by qiuxiafei** [comment]
Great job, a giant step toward a mono-workspace!
Pls also pay attention to internal build~

**Pull request merged**